In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
df = pd.read_pickle('/Users/arlettedeldique/Desktop/Arlette_BA_cleane_Notebooks/df_with_metadata.pkl')

In [3]:
df.head()

Original File Name  Genus          LABEL  \
0  001l_Culex_pipiens_Test.tif  Culex  Culex pipiens   
1  002r_Culex_pipiens_Test.tif  Culex  Culex pipiens   
2  003l_Culex_pipiens_Test.tif  Culex  Culex pipiens   
3  005l_Culex_pipiens_Test.tif  Culex  Culex pipiens   
4  005r_Culex_pipiens_Test.tif  Culex  Culex pipiens   

                                     processed_image        x1        x2  \
0  [[[0, 0, 0], [0, 0, 0], [0, 0, 0], [0, 0, 0], ...  0.350000  0.665625   
1  [[[0, 0, 0], [0, 0, 0], [0, 0, 0], [0, 0, 0], ...  0.353125  0.681250   
2  [[[0, 0, 0], [0, 0, 0], [0, 0, 0], [0, 0, 0], ...  0.326562  0.626563   
3  [[[0, 0, 0], [0, 0, 0], [0, 0, 0], [0, 0, 0], ...  0.334375  0.620313   
4  [[[0, 0, 0], [0, 0, 0], [0, 0, 0], [0, 0, 0], ...  0.339062  0.632812   

         x3        x4        x5        x6  ...           File Name base  \
0  0.948438  0.970313  0.976562  0.967187  ...  001l_Culex_pipiens_Test   
1  0.946875  0.965625  0.975000  0.973437  ...  002r_Culex_pipiens_Test   
2  0.931250  0.950000  0.956250  0.950000  ...  003l_Culex_pipiens_Test   
3  0.931250  0.956250  0.960938  0.956250  ...  005l_Culex_pipiens_Test   
4  0.948438  0.970313  0.975000  0.967187  ...  005r_Culex_pipiens_Test   

   Capture location  Capture location - LAT  Capture location - LONG  \
0               NaN                     NaN                      NaN   
1               NaN                     NaN                      NaN   
2               NaN                     NaN                      NaN   
3               NaN                     NaN                      NaN   
4               NaN                     NaN                      NaN   

   Capture method  Date of Sample Collection                       Device  \
0             NaN                        NaN  olympus sz61 + olympus dp23   
1             NaN                        NaN  olympus sz61 + olympus dp23   
2             NaN                        NaN  olympus sz61 + olympus dp23   
3             NaN                        NaN  olympus sz61 + olympus dp23   
4             NaN                        NaN  olympus sz61 + olympus dp23   

   Method of Identification  Project  Sex  
0                       NaN      NaN    F  
1                       NaN      NaN    F  
2                       NaN      NaN    F  
3                       NaN      NaN    F  
4                       NaN      NaN    F  

[5 rows x 53 columns]

In [4]:
df.columns

Index(['Original File Name', 'Genus', 'LABEL', 'processed_image', 'x1', 'x2',
       'x3', 'x4', 'x5', 'x6', 'x7', 'x8', 'x9', 'x10', 'x11', 'x12', 'x13',
       'x14', 'x15', 'x16', 'x17', 'x18', 'y1', 'y2', 'y3', 'y4', 'y5', 'y6',
       'y7', 'y8', 'y9', 'y10', 'y11', 'y12', 'y13', 'y14', 'y15', 'y16',
       'y17', 'y18', 'Wing_Side', 'File Name clean', 'Image Path',
       'File Name base', 'Capture location', 'Capture location - LAT',
       'Capture location - LONG', 'Capture method',
       'Date of Sample Collection', 'Device', 'Method of Identification',
       'Project', 'Sex'],
      dtype='object')

In [5]:
df.shape

(4935, 53)

In [6]:
#analyse für methoden
df['Genus'].value_counts()

Genus
Aedes             2650
Culex             1998
Coquillettidia      88
Anopheles           87
Culiseta            61
Armigeres           38
Mansonia             7
Tripteroides         6
Name: count, dtype: int64

Aedes             2650
Anopheles           87
Armigeres           38
Coquillettidia      88
Culex             1998
Culiseta            61
Mansonia             7
Tripteroides         6


In [7]:
df['LABEL'].unique()

array(['Culex pipiens', 'Aedes detritus-coluzzo', 'Culiseta annulata',
       'Culex modestus', 'Coquillettidia richiardii',
       'Culiseta morsitans', 'Aedes aegypti', 'Aedes albopictus',
       'Aedes koreicus', 'Aedes japonicus', 'Aedes annulipes',
       'Aedes caspius', 'Aedes cinerus-geminus', 'Aedes communis-punctor',
       'Aedes geniculatus', 'Aedes rossicus', 'Aedes rusticus',
       'Aedes sticticus', 'Aedes vexans', 'Anopheles claviger',
       'Anopheles maculipennis', 'Anopheles plumbeus', 'Culex territans',
       'Culex torrentium', 'Armigeres subalbatus', 'Culex vishnui',
       'Mansonia indiana', 'Aedes ostentatio', 'Aedes thailandensis',
       'Culex brevipalpis', 'Aedes amesii', 'Anopheles aconitus',
       'Culex gelidus', 'Culex bitaeniorhynchus',
       'Coquillettidia crassipes', 'Aedes vittatus', 'Anopheles sinensis',
       'Tripteroides aranoides', 'Anopheles tessellatus',
       'Armigeres moultoni', 'Culex fuscocephala', 'Tripteroides affinis',
       

In [8]:
Species = df['LABEL'].value_counts()

In [9]:
Species.tolist

<bound method IndexOpsMixin.tolist of LABEL
Culex pipiens                1507
Aedes aegypti                 915
Aedes detritus-coluzzo        427
Aedes japonicus               368
Aedes koreicus                344
Aedes albopictus              317
Culex vishnui                 252
Culex torrentium              166
Coquillettidia richiardii      58
Aedes communis-punctor         56
Aedes annulipes                38
Armigeres subalbatus           37
Culiseta morsitans             36
Aedes geniculatus              34
Aedes cinerus-geminus          30
Aedes sticticus                30
Anopheles plumbeus             30
Coquillettidia crassipes       30
Aedes vexans                   30
Anopheles claviger             28
Culiseta annulata              25
Culex territans                22
Culex gelidus                  21
Anopheles maculipennis         17
Aedes caspius                  17
Aedes rusticus                 15
Aedes rossicus                 14
Culex fuscocephala             14
Aede

In [10]:
len(Species)

46

In [11]:
df['Wing_Side'].value_counts()

Wing_Side
R    2724
L    2211
Name: count, dtype: int64

In [12]:
df['Capture location'].value_counts()

Capture location
Hamburg, Germany               1428
Rio de Janeiro, Brazil          635
MISSING VALUE                   429
Conet-en-Roussillon, France     367
Ho Chi Minh, Vietnam            261
Heidelberg, Germany             241
Wiesbaden, Germany              221
Southern Germany                213
Thua Thien Hue, Vietnam          94
Dak Lak, Vietnam                 80
Weinheim, Germany                76
Grau-de-Roi, France              60
Varel, Germany                   54
Tay Ninh, Vietnam                49
Ha Noi, Vietnam                  45
Hung Yen, Vietnam                43
Inheiden, Germany                39
Regensburg, Germany              38
Landau, Germany                  37
Rackwitz,Germany                 35
Lindau, Germany                  34
Zschepplin, Germany              34
Elsterwerda, Germany             34
Kassel, Germany                  34
Waldenbuch, Germany              30
Ravensburg, Germany              30
Karlsruhe, Germany               29
Bad Schmied

In [13]:
df['Capture method'].value_counts()

Capture method
laboratory-colony, bnitm hamburg                      1525
bg-pro                                                 572
egg-raft-collection                                    479
 egg-raft collection                                   427
laboratory colony, fiocruz, rio de janeiro, brazil     401
bg-sentinel                                            344
ovitrap                                                234
aspirator                                               58
Name: count, dtype: int64

In [14]:
df['Date of Sample Collection'].value_counts()

Date of Sample Collection
MISSING VALUE          1220
2022-07-15 00:00:00     759
2022-05-01 00:00:00     635
2022-11-26 00:00:00     221
13.08.2021               63
                       ... 
2017-07-24 00:00:00       1
2017-08-10 00:00:00       1
2018-08-26 00:00:00       1
2018-07-25 00:00:00       1
23.07.2021                1
Name: count, Length: 167, dtype: int64

In [15]:
df['Device'].value_counts()

Device
olympus sz61 + olympus dp23    3866
macrolens + iphone se           613
leica-m205c                     429
Name: count, dtype: int64

In [16]:
df['Method of Identification'].value_counts()

Method of Identification
Laboratory-Colony, BNITM Hamburg    1525
morphology                           935
COI gene barcoding                   659
qPCR CQ11 and ACE2 genes             477
ITS2 gene barcoding                   17
Name: count, dtype: int64

In [17]:
df['Project'].value_counts()

Project
ConVector                           937
pipiens-complex-study               759
aegypti-wolbachia-study             635
NEED                                634
landmark-mosquito-identification    429
EID-MED + BNITM Colab               427
vietnam-mosquito-collection-2022    398
landmark-aedes-collection           281
vietnam-mosquito-collection-2023    174
AIDA                                 98
Name: count, dtype: int64

In [18]:
df['Sex'].value_counts()

Sex
F    4199
M     709
Name: count, dtype: int64

einen dataframe ohne bilder, die mit dem iPhone aufgenommen wurden 

In [19]:
df_without_iphone = df[df["Device"] != "macrolens + iphone se"]

In [20]:
df_without_iphone.shape

(4322, 53)

In [21]:
df_without_iphone.columns

Index(['Original File Name', 'Genus', 'LABEL', 'processed_image', 'x1', 'x2',
       'x3', 'x4', 'x5', 'x6', 'x7', 'x8', 'x9', 'x10', 'x11', 'x12', 'x13',
       'x14', 'x15', 'x16', 'x17', 'x18', 'y1', 'y2', 'y3', 'y4', 'y5', 'y6',
       'y7', 'y8', 'y9', 'y10', 'y11', 'y12', 'y13', 'y14', 'y15', 'y16',
       'y17', 'y18', 'Wing_Side', 'File Name clean', 'Image Path',
       'File Name base', 'Capture location', 'Capture location - LAT',
       'Capture location - LONG', 'Capture method',
       'Date of Sample Collection', 'Device', 'Method of Identification',
       'Project', 'Sex'],
      dtype='object')

In [22]:
cols = [
    'File Name clean', 'Genus', 'LABEL', 'processed_image',
    'x1', 'x2', 'x3', 'x4', 'x5', 'x6', 'x7', 'x8', 'x9', 'x10', 'x11', 'x12', 'x13',
    'x14', 'x15', 'x16', 'x17', 'x18',
    'y1', 'y2', 'y3', 'y4', 'y5', 'y6', 'y7', 'y8', 'y9', 'y10', 'y11', 'y12', 'y13',
    'y14', 'y15', 'y16', 'y17', 'y18'
]

df_without_iphone_ml = df_without_iphone[cols].reset_index(drop=True)


In [23]:
df_without_iphone_ml.head()

File Name clean  Genus          LABEL  \
0  001l_Culex_pipiens_Test  Culex  Culex pipiens   
1  002r_Culex_pipiens_Test  Culex  Culex pipiens   
2  003l_Culex_pipiens_Test  Culex  Culex pipiens   
3  005l_Culex_pipiens_Test  Culex  Culex pipiens   
4  005r_Culex_pipiens_Test  Culex  Culex pipiens   

                                     processed_image        x1        x2  \
0  [[[0, 0, 0], [0, 0, 0], [0, 0, 0], [0, 0, 0], ...  0.350000  0.665625   
1  [[[0, 0, 0], [0, 0, 0], [0, 0, 0], [0, 0, 0], ...  0.353125  0.681250   
2  [[[0, 0, 0], [0, 0, 0], [0, 0, 0], [0, 0, 0], ...  0.326562  0.626563   
3  [[[0, 0, 0], [0, 0, 0], [0, 0, 0], [0, 0, 0], ...  0.334375  0.620313   
4  [[[0, 0, 0], [0, 0, 0], [0, 0, 0], [0, 0, 0], ...  0.339062  0.632812   

         x3        x4        x5        x6  ...        y9       y10       y11  \
0  0.948438  0.970313  0.976562  0.967187  ...  0.703125  0.765625  0.571875   
1  0.946875  0.965625  0.975000  0.973437  ...  0.618750  0.681250  0.471875   
2  0.931250  0.950000  0.956250  0.950000  ...  0.671875  0.721875  0.496875   
3  0.931250  0.956250  0.960938  0.956250  ...  0.643750  0.706250  0.509375   
4  0.948438  0.970313  0.975000  0.967187  ...  0.656250  0.709375  0.493750   

        y12       y13       y14      y15       y16       y17       y18  
0  0.550000  0.509375  0.506250  0.47500  0.446875  0.425000  0.550000  
1  0.450000  0.409375  0.406250  0.37500  0.337500  0.325000  0.456250  
2  0.500000  0.450000  0.453125  0.41875  0.384375  0.365625  0.509375  
3  0.490625  0.443750  0.443750  0.40625  0.368750  0.356250  0.493750  
4  0.478125  0.431250  0.440625  0.39375  0.359375  0.337500  0.481250  

[5 rows x 40 columns]

In [ ]:
df_without_iphone_ml.to_pickle('/Users/arlettedeldique/Desktop/df_without_iphone_ml.pkl')

SyntaxError: invalid syntax (1413027069.py, line 1)

analyse df without iphone + lens pictures 

In [26]:
df_without_iphone.shape

(4322, 53)

In [27]:
df_without_iphone['Genus'].value_counts()

Genus
Aedes             2037
Culex             1998
Coquillettidia      88
Anopheles           87
Culiseta            61
Armigeres           38
Mansonia             7
Tripteroides         6
Name: count, dtype: int64

In [28]:
Species = df_without_iphone['LABEL'].value_counts()
Species.tolist

<bound method IndexOpsMixin.tolist of LABEL
Culex pipiens                1507
Aedes aegypti                 742
Aedes detritus-coluzzo        427
Culex vishnui                 252
Aedes japonicus               215
Aedes koreicus                207
Aedes albopictus              167
Culex torrentium              166
Coquillettidia richiardii      58
Aedes communis-punctor         56
Aedes annulipes                38
Armigeres subalbatus           37
Culiseta morsitans             36
Aedes geniculatus              34
Aedes cinerus-geminus          30
Aedes sticticus                30
Anopheles plumbeus             30
Coquillettidia crassipes       30
Aedes vexans                   30
Anopheles claviger             28
Culiseta annulata              25
Culex territans                22
Culex gelidus                  21
Anopheles maculipennis         17
Aedes caspius                  17
Aedes rusticus                 15
Aedes rossicus                 14
Culex fuscocephala             14
Aede

In [29]:
len(Species)

46

In [30]:
df_without_iphone['Wing_Side'].value_counts()

Wing_Side
L    2211
R    2111
Name: count, dtype: int64

In [31]:
df_without_iphone['Capture method'].value_counts()

Capture method
laboratory-colony, bnitm hamburg                      912
bg-pro                                                572
egg-raft-collection                                   479
 egg-raft collection                                  427
laboratory colony, fiocruz, rio de janeiro, brazil    401
bg-sentinel                                           344
ovitrap                                               234
aspirator                                              58
Name: count, dtype: int64

In [32]:
df_without_iphone['Date of Sample Collection'].value_counts()

Date of Sample Collection
2022-07-15 00:00:00    759
2022-05-01 00:00:00    635
MISSING VALUE          607
2022-11-26 00:00:00    221
13.08.2021              63
                      ... 
2017-07-24 00:00:00      1
2017-08-10 00:00:00      1
2018-08-26 00:00:00      1
2018-07-25 00:00:00      1
23.07.2021               1
Name: count, Length: 167, dtype: int64

In [33]:
df_without_iphone['Device'].value_counts()

Device
olympus sz61 + olympus dp23    3866
leica-m205c                     429
Name: count, dtype: int64

In [34]:
df_without_iphone['Method of Identification'].value_counts()

Method of Identification
morphology                          935
Laboratory-Colony, BNITM Hamburg    912
COI gene barcoding                  659
qPCR CQ11 and ACE2 genes            477
ITS2 gene barcoding                  17
Name: count, dtype: int64

In [35]:
df_without_iphone['Project'].value_counts()

Project
pipiens-complex-study               759
aegypti-wolbachia-study             635
NEED                                634
landmark-mosquito-identification    429
EID-MED + BNITM Colab               427
vietnam-mosquito-collection-2022    398
ConVector                           324
landmark-aedes-collection           281
vietnam-mosquito-collection-2023    174
AIDA                                 98
Name: count, dtype: int64

In [36]:
df_without_iphone['Sex'].value_counts()

Sex
F    3586
M     709
Name: count, dtype: int64